### PR TITLE
doc: workforce identity federation for pluggable auth

### PR DESCRIPTION
**Pull request by BigTailWolf**
Adding documents

### CODE REVIEW

File: docs/user-guide.rst [hunk 1]
```diff
@@ -544,6 +544,246 @@ For AWS providers, use :meth:`aws.Credentials.from_info
...
+
+`Workforce identity federation`_ lets you use an external identify provider
+(IdP) to authenticate and authorize a workforce—a group of users, such as
+employees, partners, and contractors—using IAM, so that the users can access
+Google Cloud services. Workforce identity federation extends Google Cloud's
+identity capabilities to support syncless, attribute-based single sign on.
+
+With workforce identity federation, your workforce can access Google Cloud
+resources using an external identity provider (IdP) that supports OpenID
+Connect (OIDC) or SAML 2.0 such as Azure Active Directory (Azure AD), Active
```
Comment (sai-sunder-s) [inline docs/user-guide.rst:558]: we can add AWS as well here?

File: docs/user-guide.rst [hunk 2]
```diff
@@ -544,6 +544,246 @@ For AWS providers, use :meth:`aws.Credentials.from_info
...
+* ``$WORKFORCE_POOL_ID``: The workforce pool ID.
+* ``$PROVIDER_ID``: The provider ID.
+* ``$PATH_TO_SAML_ASSERTION``: The file path used to retrieve the
+  base64-encoded SAML assertion.
+* ``$WORKFORCE_POOL_USER_PROJECT``: The project number associated with the
+  `workforce pools user project`_.
+
+These commands generate the configuration file in the specified output file.
+
+URL-sourced credentials For URL-sourced credentials, a local server needs to
```
Comment (sai-sunder-s) [inline docs/user-guide.rst:638]: did you intend this to be **URL-sourced credentials** For URL-sourced credentials?

Same for file sourced

File: docs/user-guide.rst (hunk 1 above)
Comment (BigTailWolf) [reply docs/user-guide.rst:558]: AWS is not OIDC IIRC

File: docs/user-guide.rst (hunk 2 above)
Comment (BigTailWolf) [reply docs/user-guide.rst:638]: Took a look at the source doc in Java library. That was a title which didn't outlined. I'll fix here

File: docs/user-guide.rst (hunk 1 above)
Comment (sai-sunder-s) [reply docs/user-guide.rst:558]: But AWS is supported right? We have AWS specific WIF code

Comment (lsirac) [reply docs/user-guide.rst:558]: Workforce identity federation does not support AWS.

**Review by lsirac — APPROVED** [review]

**Review by sai-sunder-s — APPROVED** [review]

**Pull request merged**